templates/frontend/objects/galley_link.tpl

```diff
@@ -48,8 +48,21 @@
 	{/if}
 {/if}
 
+{* Revision? *}
+{if $isPreviousRevision}
+	{assign var=path value=$parentId|to_array:$submissionRevision:$galley->getBestGalleyId()}
+	{assign var=label value=$galley->getGalleyLabel()}
+{else}
+	{assign var=path value=$parentId|to_array:$galley->getBestGalleyId()}
+	{assign var=label value=$galley->getGalleyLabel()}
+{/if}
+
 {* Don't be frightened. This is just a link *}
-<a class="obj_galley_link {$type}{if $restricted} restricted{/if}" href="{url page=$page op="view" path=$parentId|to_array:$galley->getBestGalleyId()}">
+{if $isPreviousRevision}
+	<a class="obj_galley_link {$type}{if $restricted} restricted{/if}" href="{url page=$page op="version" path=$path}">
+{else}
+	<a class="obj_galley_link {$type}{if $restricted} restricted{/if}" href="{url page=$page op="view" path=$path}">
+{/if}
```

Comment:
maybe to set the operation as a variable above, so that here only one line of the code can be used?